Lib/test/test_typing.py

```diff
@@ -2734,6 +2734,17 @@ def test_get_type_hints_ClassVar(self):
                          {'z': ClassVar[CSub], 'y': int, 'b': int,
                           'x': ClassVar[Optional[B]]})
         self.assertEqual(gth(G), {'lst': ClassVar[List[T]]})
+        
+    def test_get_type_hints_retains_forward_equality(self):
+        fr = typing.ForwardRef('int')
+        hsh = hash(fr)
+        
+        def foo(a: fr):
+            pass
+        
+        gth(foo)
+        self.assertEqual(fr, typing.ForwardRef('int'))
+        self.assertEqual(hash(fr), hsh)
```

Comment:
Please add few more tests, e.g. checking also equality and hashes between references that got evaluated to different things etc.